xrpl/utils/mptoken_metadata.py

```
@@ -0,0 +1,397 @@
+        This utility is the counterpart to the encoding function and handles
+        field name expansion.
+    """
+    if bool(HEX_REGEX.fullmatch(input_hex)) is False:
+        raise ValueError("MPTokenMetadata must be in hex format.")
+
+    try:
+        json_metadata = json.loads(hex_to_str(input_hex))
+    except Exception as e:
+        raise ValueError(
+            f"MPTokenMetadata is not properly formatted as JSON - {e}"
+        ) from e
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)